rubix-prestosql/src/test/java/com/qubole/rubix/prestosql/TestingNodeManager.java

```diff
@@ -0,0 +1,287 @@
+/**
+ * Copyright (c) 2019. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.prestosql;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.qubole.rubix.common.utils.ClusterUtil;
+import io.prestosql.spi.HostAddress;
+import io.prestosql.spi.Node;
+import io.prestosql.spi.NodeManager;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.net.*;
+import java.util.*;
+
+import static java.util.Objects.requireNonNull;
+
+public class TestingNodeManager implements NodeManager {
+  private static Log LOG = LogFactory.getLog(TestingNodeManager.class);
+  private static final int DEFAULT_SERVER_PORT = 8081;
+  private static final String DEFAULT_USER = "rubix";
+  public static final String SERVER_PORT_CONF_KEY = "caching.fs.presto-server-port";
+
+  private final String serverAddress;
+  private final Node currentNode;
+  private final int serverPort;
+
+  public TestingNodeManager(Configuration conf, Node currentNode)
```

Comment:
I guess you wont need this once you have Sync and Async implementations of PrestoClusterManager